.github/workflows/test.yml

```
@@ -32,3 +32,44 @@ jobs:
     - run: python -m pip install .
     - run: python -m pip install pytest
     - run: python -m pytest
+
+  test-aarch64:
+    name: "Test ${{ matrix.pyver }} aarch64"
+    strategy:
+      matrix:
+        pyver: [cp39-cp39]
+        arch: [aarch64]
+      fail-fast: false
+    runs-on: ubuntu-latest
+    env:
+      py: /opt/python/${{ matrix.pyver }}/bin/python
+      img: quay.io/pypa/manylinux2014_${{ matrix.arch }}
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+    - name: Set up QEMU
+      id: qemu
+      uses: docker/setup-qemu-action@v1
+    - name: Install tools
+      run: |
+        docker run --rm -v ${{ github.workspace }}:/ws:rw --workdir=/ws \
+          ${{ env.img }} ${{ env.py }} -m pip install -U  pip build
+    - name: Install dependencies
+      run: |
+            docker run --rm -v ${{ github.workspace }}:/ws:rw --workdir=/ws \
+            ${{ env.img }} \
+            bash -exc '${{ env.py }} -m venv .env && \
+            source .env/bin/activate && \
+            python -m pip install --upgrade pip setuptools wheel && \
+            deactivate'
+    - name: Test with pytest
+      run: |
+          docker run --rm -v ${{ github.workspace }}:/ws:rw --workdir=/ws \
+            ${{ env.img }} \
+            bash -exc '\
+            source .env/bin/activate && \
+            python -m pip install . && \
+            pip install pytest && \
+            python -m pytest && \
+            deactivate'
```

```suggestion
      run: >-
          docker run --rm -v ${{ github.workspace }}:/ws:rw --workdir=/ws
            ${{ env.img }}
            bash -exc '
              source .env/bin/activate &&
              python -m pip install . &&
              pip install pytest &&
              python -m pytest &&
              deactivate'
```
Even more so here, where they hide the quotes being used on the command, IMO.

---

Isn't there a testing extra? If that's the case, you should do `python -m pip install .[test]`, rather than manually installing pytest. If we wanted to add a pytest plugin in the future, for example, or restrict pytest, etc, it's better to use a centrally managed extra.

Also, you don't need to activate and deactivate the environment, you can just call `.env/bin/python -m ...`

Please name virtual environments `.venv` instead of `.env`. That's actually becoming a standard name; the Python launcher for Unix even supports it directly.

---

Also, PEP 518 means you don't have to install/upgrade `setuptools` & `wheel` anymore, they are specified in pyproject.toml.